src/ImageService.Broker/Consumers/ImageUser/CreateImagesUserConsumer.cs

```diff
@@ -0,0 +1,56 @@
+﻿using LT.DigitalOffice.ImageService.Data.Interfaces;
+using LT.DigitalOffice.ImageService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Models;
+using LT.DigitalOffice.Models.Broker.Requests.Image;
+using LT.DigitalOffice.Models.Broker.Responses.Image;
+using MassTransit;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ImageService.Broker.Consumers.ImageUser
+{
+    public class CreateImagesUserConsumer : IConsumer<ICreateImagesUserRequest>
+    {
+        private readonly IImageUserRepository _repository;
+        private readonly IDbImageUserMapper _mapper;
+
+        public CreateImagesUserConsumer(IImageUserRepository repository, IDbImageUserMapper mapper)
+        {
+            _repository = repository;
+            _mapper = mapper;
+        }
+        public async Task Consume(ConsumeContext<ICreateImagesUserRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(CreateImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<ICreateImagesResponse>>(response);
+        }
+
+        private object CreateImages(ICreateImagesUserRequest request)
+        {
+            List<CreateImageData> createImage = request.CreateImagesData;
+            List<DbImagesUser> images = new();
+
+            foreach (CreateImageData x in createImage)
+            {
+                images.AddRange(CreateHQ(x));
```

Comment:
a very strange method. Every iteration, you turn to the method that creates the list and then put this list into the list

Reply:
Replaced with the normal method